src/OVAL/oval_session.c

```diff
@@ -0,0 +1,463 @@
+/*
+ * Copyright 2015 Red Hat Inc., Durham, North Carolina.
+ * All Rights Reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Author:
+ * 		Michal Šrubař <msrubar@redhat.com>
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <libgen.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "common/alloc.h"
+#include "common/util.h"
+#include "common/_error.h"
+#include "common/oscapxml.h"
+#include "source/xslt_priv.h"
+#include "public/oval_agent_api.h"
+#include "public/oval_session.h"
+#include "../DS/public/ds_sds_session.h"
+#include "oscap_source.h"
+
+static const char *oscap_productname = "cpe:/a:open-scap:oscap";
+static const char *oval_results_report = "oval-results-report.xsl";
+
+struct oval_session {
+	/* Main source assigned with the main file (SDS or OVAL) */
+	struct oscap_source *source;
+	struct oval_definition_model *def_model;
+	struct oval_variable_model *var_model;
+	struct oval_results_model *res_model;
+
+	oval_agent_session_t *sess;
+	struct ds_sds_session *sds_session;
+
+	struct {
+		struct oscap_source *definitions;
+		struct oscap_source *variables;
+		struct oscap_source *directives;
+	} oval;
+
+	const char *datastream_id;
+	/* particular OVAL component if there are two OVALs in one datastream */
+	const char *component_id;
+
+	struct {
+		const char *results;
+		const char *report;
+	} export;
+
+	struct {
+		/* it's called when there is something invalid in input/output files */
+		xml_reporter xml_fn;
+	} reporter;
+
+	bool validation;
+	bool full_validation;
+};
+
+struct oval_session *oval_session_new(const char *filename)
+{
+	oscap_document_type_t scap_type;
+	struct oval_session *session;
+
+	session = (struct oval_session *) oscap_calloc(1, sizeof(struct oval_session));
```

Comment:
I think "(struct oval_session *)" is useless in C99. - You have declaration near the allocation, so you can see type of pointer without typecasting.